2024-12-03 19:17:42,529 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-03 19:17:42,529 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-03 20:59:59,368 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-03 20:59:59,368 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-03 21:08:16,396 - app.extensions - INFO - Response: 415 UNSUPPORTED MEDIA TYPE - Data: <!doctype html>
<html lang=en>
<title>415 Unsupported Media Type</title>
<h1>Unsupported Media Type</h1>
<p>Did not attempt to load JSON data because the request Content-Type was not &#39;application/json&#39;.</p>

2024-12-03 21:08:16,398 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 21:08:16] "[31m[1mPOST /api/customers/register HTTP/1.1[0m" 415 -
2024-12-03 21:09:27,545 - app.extensions - INFO - Request: POST http://127.0.0.1:5000/api/customers/register - Data: {'full_name': 'John Doe', 'username': 'johndoe', 'password': 'securepassword', 'age': 30, 'address': '123 Main St', 'gender': 'Male', 'marital_status': 'Single'}
2024-12-03 21:09:27,556 - app.app - ERROR - Exception on /api/customers/register [POST]
Traceback (most recent call last):
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/app/routes/customers.py", line 11, in register_customer
    if Customer.query.filter_by(username=data['username']).first():
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/query.py", line 2728, in first
    return self.limit(1)._iter().first()  # type: ignore
           ~~~~~~~~~~~~~~~~~~~^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/query.py", line 2827, in _iter
    result: Union[ScalarResult[_T], Result[_T]] = self.session.execute(
                                                  ~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
        params,
        ^^^^^^^
        execution_options={"_sa_orm_load_options": self.load_options},
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2362, in execute
    return self._execute_internal(
           ~~~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
    ...<4 lines>...
        _add_event=_add_event,
        ^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2235, in _execute_internal
    bind = self.get_bind(**bind_arguments)
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask_sqlalchemy/session.py", line 53, in get_bind
    engines = self._db.engines
              ^^^^^^^^^^^^^^^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask_sqlalchemy/extension.py", line 690, in engines
    raise RuntimeError(
    ...<3 lines>...
    )
RuntimeError: The current Flask app is not registered with this 'SQLAlchemy' instance. Did you forget to call 'init_app', or did you create multiple 'SQLAlchemy' instances?
2024-12-03 21:09:27,565 - app.extensions - INFO - Response: 500 INTERNAL SERVER ERROR - Data: <!doctype html>
<html lang=en>
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>

2024-12-03 21:09:27,566 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 21:09:27] "[35m[1mPOST /api/customers/register HTTP/1.1[0m" 500 -
2024-12-03 21:29:19,401 - flask_migrate - ERROR - Error: Directory migrations already exists and is not empty
2024-12-03 21:30:24,910 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-03 21:30:24,910 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-03 21:30:30,929 - app.extensions - INFO - Request: POST http://127.0.0.1:5000/api/customers/register - Data: {'full_name': 'John Doe', 'username': 'johndoe', 'password': 'securepassword', 'age': 30, 'address': '123 Main St', 'gender': 'Male', 'marital_status': 'Single'}
2024-12-03 21:30:30,931 - app.app - ERROR - Exception on /api/customers/register [POST]
Traceback (most recent call last):
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/app/routes/customers.py", line 10, in register_customer
    if Customer.query.filter_by(username=data['username']).first():
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/query.py", line 2728, in first
    return self.limit(1)._iter().first()  # type: ignore
           ~~~~~~~~~~~~~~~~~~~^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/query.py", line 2827, in _iter
    result: Union[ScalarResult[_T], Result[_T]] = self.session.execute(
                                                  ~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
        params,
        ^^^^^^^
        execution_options={"_sa_orm_load_options": self.load_options},
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2362, in execute
    return self._execute_internal(
           ~~~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
    ...<4 lines>...
        _add_event=_add_event,
        ^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2235, in _execute_internal
    bind = self.get_bind(**bind_arguments)
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask_sqlalchemy/session.py", line 53, in get_bind
    engines = self._db.engines
              ^^^^^^^^^^^^^^^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask_sqlalchemy/extension.py", line 690, in engines
    raise RuntimeError(
    ...<3 lines>...
    )
RuntimeError: The current Flask app is not registered with this 'SQLAlchemy' instance. Did you forget to call 'init_app', or did you create multiple 'SQLAlchemy' instances?
2024-12-03 21:30:30,937 - app.extensions - INFO - Response: 500 INTERNAL SERVER ERROR - Data: <!doctype html>
<html lang=en>
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>

2024-12-03 21:30:30,937 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 21:30:30] "[35m[1mPOST /api/customers/register HTTP/1.1[0m" 500 -
2024-12-03 21:47:39,004 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-03 21:47:39,005 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-03 21:48:14,901 - app.extensions - INFO - Request: POST http://127.0.0.1:5000/api/customers/register - Data: {'full_name': 'John Doe', 'username': 'johndoe', 'password': 'securepassword', 'age': 30, 'address': '123 Main St', 'gender': 'Male', 'marital_status': 'Single'}
2024-12-03 21:48:14,906 - app.app - ERROR - Exception on /api/customers/register [POST]
Traceback (most recent call last):
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/app/routes/customers.py", line 10, in register_customer
    if Customer.query.filter_by(username=data['username']).first():
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/query.py", line 2728, in first
    return self.limit(1)._iter().first()  # type: ignore
           ~~~~~~~~~~~~~~~~~~~^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/query.py", line 2827, in _iter
    result: Union[ScalarResult[_T], Result[_T]] = self.session.execute(
                                                  ~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
        params,
        ^^^^^^^
        execution_options={"_sa_orm_load_options": self.load_options},
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2362, in execute
    return self._execute_internal(
           ~~~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
    ...<4 lines>...
        _add_event=_add_event,
        ^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2235, in _execute_internal
    bind = self.get_bind(**bind_arguments)
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask_sqlalchemy/session.py", line 53, in get_bind
    engines = self._db.engines
              ^^^^^^^^^^^^^^^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask_sqlalchemy/extension.py", line 690, in engines
    raise RuntimeError(
    ...<3 lines>...
    )
RuntimeError: The current Flask app is not registered with this 'SQLAlchemy' instance. Did you forget to call 'init_app', or did you create multiple 'SQLAlchemy' instances?
2024-12-03 21:48:14,912 - app.extensions - INFO - Response: 500 INTERNAL SERVER ERROR - Data: <!doctype html>
<html lang=en>
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>

2024-12-03 21:48:14,914 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 21:48:14] "[35m[1mPOST /api/customers/register HTTP/1.1[0m" 500 -
2024-12-03 22:01:06,769 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-03 22:01:06,770 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-03 22:01:13,042 - app.extensions - INFO - Request: POST http://127.0.0.1:5000/api/customers/register - Data: {'full_name': 'John Doe', 'username': 'johndoe', 'password': 'securepassword', 'age': 30, 'address': '123 Main St', 'gender': 'Male', 'marital_status': 'Single'}
2024-12-03 22:01:13,051 - app.app - ERROR - Exception on /api/customers/register [POST]
Traceback (most recent call last):
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
    ~~~~~~~~~~~~~~~~~~~~~~~^
        cursor, str_statement, effective_parameters, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/default.py", line 941, in do_execute
    cursor.execute(statement, parameters)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such table: customer

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/app/routes/customers.py", line 10, in register_customer
    if Customer.query.filter_by(username=data['username']).first():
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/query.py", line 2728, in first
    return self.limit(1)._iter().first()  # type: ignore
           ~~~~~~~~~~~~~~~~~~~^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/query.py", line 2827, in _iter
    result: Union[ScalarResult[_T], Result[_T]] = self.session.execute(
                                                  ~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
        params,
        ^^^^^^^
        execution_options={"_sa_orm_load_options": self.load_options},
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2362, in execute
    return self._execute_internal(
           ~~~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
    ...<4 lines>...
        _add_event=_add_event,
        ^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2247, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self,
        ^^^^^
    ...<4 lines>...
        conn,
        ^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/context.py", line 305, in orm_execute_statement
    result = conn.execute(
        statement, params or {}, execution_options=execution_options
    )
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1418, in execute
    return meth(
        self,
        distilled_parameters,
        execution_options or NO_OPTIONS,
    )
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/sql/elements.py", line 515, in _execute_on_connection
    return connection._execute_clauseelement(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self, distilled_params, execution_options
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1640, in _execute_clauseelement
    ret = self._execute_context(
        dialect,
    ...<8 lines>...
        cache_hit=cache_hit,
    )
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1846, in _execute_context
    return self._exec_single_context(
           ~~~~~~~~~~~~~~~~~~~~~~~~~^
        dialect, context, statement, parameters
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1986, in _exec_single_context
    self._handle_dbapi_exception(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        e, str_statement, effective_parameters, cursor, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 2355, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
    ~~~~~~~~~~~~~~~~~~~~~~~^
        cursor, str_statement, effective_parameters, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/default.py", line 941, in do_execute
    cursor.execute(statement, parameters)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: customer
[SQL: SELECT customer.id AS customer_id, customer.full_name AS customer_full_name, customer.username AS customer_username, customer.password AS customer_password, customer.age AS customer_age, customer.address AS customer_address, customer.gender AS customer_gender, customer.marital_status AS customer_marital_status, customer.wallet_balance AS customer_wallet_balance 
FROM customer 
WHERE customer.username = ?
 LIMIT ? OFFSET ?]
[parameters: ('johndoe', 1, 0)]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
2024-12-03 22:01:13,059 - app.extensions - INFO - Response: 500 INTERNAL SERVER ERROR - Data: <!doctype html>
<html lang=en>
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>

2024-12-03 22:01:13,059 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:01:13] "[35m[1mPOST /api/customers/register HTTP/1.1[0m" 500 -
2024-12-03 22:09:47,683 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-03 22:09:47,684 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-03 22:09:53,344 - app - ERROR - Exception on /api/customers/register [POST]
Traceback (most recent call last):
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
    ~~~~~~~~~~~~~~~~~~~~~~~^
        cursor, str_statement, effective_parameters, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/default.py", line 941, in do_execute
    cursor.execute(statement, parameters)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such table: customer

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/app/routes/customers.py", line 10, in register_customer
    if Customer.query.filter_by(username=data['username']).first():
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/query.py", line 2728, in first
    return self.limit(1)._iter().first()  # type: ignore
           ~~~~~~~~~~~~~~~~~~~^^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/query.py", line 2827, in _iter
    result: Union[ScalarResult[_T], Result[_T]] = self.session.execute(
                                                  ~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
        params,
        ^^^^^^^
        execution_options={"_sa_orm_load_options": self.load_options},
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2362, in execute
    return self._execute_internal(
           ~~~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
    ...<4 lines>...
        _add_event=_add_event,
        ^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2247, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self,
        ^^^^^
    ...<4 lines>...
        conn,
        ^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/orm/context.py", line 305, in orm_execute_statement
    result = conn.execute(
        statement, params or {}, execution_options=execution_options
    )
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1418, in execute
    return meth(
        self,
        distilled_parameters,
        execution_options or NO_OPTIONS,
    )
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/sql/elements.py", line 515, in _execute_on_connection
    return connection._execute_clauseelement(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self, distilled_params, execution_options
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1640, in _execute_clauseelement
    ret = self._execute_context(
        dialect,
    ...<8 lines>...
        cache_hit=cache_hit,
    )
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1846, in _execute_context
    return self._exec_single_context(
           ~~~~~~~~~~~~~~~~~~~~~~~~~^
        dialect, context, statement, parameters
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1986, in _exec_single_context
    self._handle_dbapi_exception(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        e, str_statement, effective_parameters, cursor, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 2355, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
    ~~~~~~~~~~~~~~~~~~~~~~~^
        cursor, str_statement, effective_parameters, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/jihadmobarak/Desktop/Ecommerce_435L/venv/lib/python3.13/site-packages/sqlalchemy/engine/default.py", line 941, in do_execute
    cursor.execute(statement, parameters)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: customer
[SQL: SELECT customer.id AS customer_id, customer.full_name AS customer_full_name, customer.username AS customer_username, customer.password AS customer_password, customer.age AS customer_age, customer.address AS customer_address, customer.gender AS customer_gender, customer.marital_status AS customer_marital_status, customer.wallet_balance AS customer_wallet_balance 
FROM customer 
WHERE customer.username = ?
 LIMIT ? OFFSET ?]
[parameters: ('johndoe', 1, 0)]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
2024-12-03 22:09:53,352 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:09:53] "[35m[1mPOST /api/customers/register HTTP/1.1[0m" 500 -
2024-12-03 22:13:42,371 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-03 22:13:42,372 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-03 22:14:03,058 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:14:03] "[35m[1mPOST /api/customers/register HTTP/1.1[0m" 201 -
2024-12-03 22:17:23,708 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:17:23] "GET /api/customers/ HTTP/1.1" 200 -
2024-12-03 22:19:48,705 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:19:48] "GET /api/customers/johndoe HTTP/1.1" 200 -
2024-12-03 22:21:19,541 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:21:19] "PUT /api/customers/johndoe HTTP/1.1" 200 -
2024-12-03 22:22:13,511 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:22:13] "GET /api/customers/johndoe HTTP/1.1" 200 -
2024-12-03 22:23:14,693 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:23:14] "POST /api/customers/johndoe/charge HTTP/1.1" 200 -
2024-12-03 22:24:37,036 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:24:37] "POST /api/customers/johndoe/deduct HTTP/1.1" 200 -
2024-12-03 22:25:34,134 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:25:34] "[35m[1mGET /api/customers/health HTTP/1.1[0m" 500 -
2024-12-03 22:27:26,596 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-03 22:27:26,596 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-03 22:27:31,447 - werkzeug - INFO - 127.0.0.1 - - [03/Dec/2024 22:27:31] "GET /api/customers/health HTTP/1.1" 200 -
2024-12-04 22:07:32,014 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-04 22:07:32,015 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-04 22:09:42,698 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:09:42] "[31m[1mPOST /api/customers/register HTTP/1.1[0m" 400 -
2024-12-04 22:09:55,745 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:09:55] "[31m[1mPOST /api/customers/register HTTP/1.1[0m" 400 -
2024-12-04 22:10:16,771 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:10:16] "[35m[1mPOST /api/customers/register HTTP/1.1[0m" 201 -
2024-12-04 22:11:23,859 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:11:23] "GET /api/customers/ HTTP/1.1" 200 -
2024-12-04 22:11:44,500 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:11:44] "GET /api/customers/jihad7 HTTP/1.1" 200 -
2024-12-04 22:12:15,002 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:12:15] "[31m[1mPUT /api/customers/jihad7 HTTP/1.1[0m" 400 -
2024-12-04 22:12:23,762 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:12:23] "PUT /api/customers/jihad7 HTTP/1.1" 200 -
2024-12-04 22:12:38,301 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:12:38] "DELETE /api/customers/johndoe HTTP/1.1" 200 -
2024-12-04 22:13:38,260 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:13:38] "POST /api/customers/jihad7/charge HTTP/1.1" 200 -
2024-12-04 22:13:54,888 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:13:54] "POST /api/customers/jihad7/deduct HTTP/1.1" 200 -
2024-12-04 22:14:04,342 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:14:04] "GET /api/customers/health HTTP/1.1" 200 -
2024-12-04 22:23:53,341 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:23:53] "[33mGET /api/sales/goods HTTP/1.1[0m" 404 -
2024-12-04 22:24:28,560 - werkzeug - INFO - 127.0.0.1 - - [04/Dec/2024 22:24:28] "[33mGET /api/sales/goods HTTP/1.1[0m" 404 -
2024-12-05 13:25:42,195 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-05 13:25:42,196 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-05 13:46:59,728 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-05 13:46:59,730 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-05 13:47:07,489 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:47:07] "GET /api/sales/goods HTTP/1.1" 200 -
2024-12-05 13:49:30,665 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:49:30] "GET /api/sales/goods/Laptop HTTP/1.1" 200 -
2024-12-05 13:50:59,708 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:50:59] "[33mPOST /api/sales/purchase HTTP/1.1[0m" 404 -
2024-12-05 13:51:11,406 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:51:11] "[33mPOST /api/sales/purchase HTTP/1.1[0m" 404 -
2024-12-05 13:51:26,678 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:51:26] "GET /api/customers/ HTTP/1.1" 200 -
2024-12-05 13:51:35,652 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:51:35] "[35m[1mPOST /api/customers/register HTTP/1.1[0m" 201 -
2024-12-05 13:51:44,020 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:51:44] "[31m[1mPOST /api/sales/purchase HTTP/1.1[0m" 400 -
2024-12-05 13:51:52,352 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:51:52] "[31m[1mPOST /api/sales/purchase HTTP/1.1[0m" 400 -
2024-12-05 13:52:14,239 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:52:14] "[31m[1mPOST /api/sales/purchase HTTP/1.1[0m" 400 -
2024-12-05 13:52:35,471 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:52:35] "POST /api/customers/jihad7/charge HTTP/1.1" 200 -
2024-12-05 13:52:49,102 - pika.adapters.utils.connection_workflow - INFO - Pika version 1.3.2 connecting to ('::1', 5672, 0, 0)
2024-12-05 13:52:49,103 - pika.adapters.utils.io_services_utils - ERROR - Socket failed to connect: <socket.socket fd=12, family=30, type=1, proto=6, laddr=('::1', 60468, 0, 0)>; error=61 (Connection refused)
2024-12-05 13:52:49,103 - pika.adapters.utils.connection_workflow - ERROR - TCP Connection attempt failed: ConnectionRefusedError(61, 'Connection refused'); dest=(<AddressFamily.AF_INET6: 30>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::1', 5672, 0, 0))
2024-12-05 13:52:49,103 - pika.adapters.utils.connection_workflow - ERROR - AMQPConnector - reporting failure: AMQPConnectorSocketConnectError: ConnectionRefusedError(61, 'Connection refused')
2024-12-05 13:52:49,103 - pika.adapters.utils.connection_workflow - INFO - Pika version 1.3.2 connecting to ('127.0.0.1', 5672)
2024-12-05 13:52:49,103 - pika.adapters.utils.io_services_utils - INFO - Socket connected: <socket.socket fd=12, family=2, type=1, proto=6, laddr=('127.0.0.1', 60469), raddr=('127.0.0.1', 5672)>
2024-12-05 13:52:49,103 - pika.adapters.utils.connection_workflow - INFO - Streaming transport linked up: (<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x105858050>, _StreamingProtocolShim: <SelectConnection PROTOCOL transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x105858050> params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>).
2024-12-05 13:52:49,232 - pika.adapters.utils.connection_workflow - INFO - AMQPConnector - reporting success: <SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x105858050> params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>
2024-12-05 13:52:49,233 - pika.adapters.utils.connection_workflow - INFO - AMQPConnectionWorkflow - reporting success: <SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x105858050> params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>
2024-12-05 13:52:49,233 - pika.adapters.blocking_connection - INFO - Connection workflow succeeded: <SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x105858050> params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>
2024-12-05 13:52:49,233 - pika.adapters.blocking_connection - INFO - Created channel=1
2024-12-05 13:52:49,250 - pika.adapters.blocking_connection - INFO - Closing connection (200): Normal shutdown
2024-12-05 13:52:49,250 - pika.channel - INFO - Closing channel (200): 'Normal shutdown' on <Channel number=1 OPEN conn=<SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x105858050> params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>>
2024-12-05 13:52:49,251 - pika.channel - INFO - Received <Channel.CloseOk> on <Channel number=1 CLOSING conn=<SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x105858050> params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>>
2024-12-05 13:52:49,251 - pika.connection - INFO - Closing connection (200): 'Normal shutdown'
2024-12-05 13:52:49,252 - pika.adapters.utils.io_services_utils - INFO - Aborting transport connection: state=1; <socket.socket fd=12, family=2, type=1, proto=6, laddr=('127.0.0.1', 60469), raddr=('127.0.0.1', 5672)>
2024-12-05 13:52:49,252 - pika.adapters.utils.io_services_utils - INFO - _AsyncTransportBase._initate_abort(): Initiating abrupt asynchronous transport shutdown: state=1; error=None; <socket.socket fd=12, family=2, type=1, proto=6, laddr=('127.0.0.1', 60469), raddr=('127.0.0.1', 5672)>
2024-12-05 13:52:49,252 - pika.adapters.utils.io_services_utils - INFO - Deactivating transport: state=1; <socket.socket fd=12, family=2, type=1, proto=6, laddr=('127.0.0.1', 60469), raddr=('127.0.0.1', 5672)>
2024-12-05 13:52:49,252 - pika.connection - INFO - AMQP stack terminated, failed to connect, or aborted: opened=True, error-arg=None; pending-error=ConnectionClosedByClient: (200) 'Normal shutdown'
2024-12-05 13:52:49,252 - pika.connection - INFO - Stack terminated due to ConnectionClosedByClient: (200) 'Normal shutdown'
2024-12-05 13:52:49,252 - pika.adapters.utils.io_services_utils - INFO - Closing transport socket and unlinking: state=3; <socket.socket fd=12, family=2, type=1, proto=6, laddr=('127.0.0.1', 60469), raddr=('127.0.0.1', 5672)>
2024-12-05 13:52:49,252 - pika.adapters.blocking_connection - INFO - User-initiated close: result=BlockingConnection__OnClosedArgs(connection=<SelectConnection CLOSED transport=None params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>, error=ConnectionClosedByClient: (200) 'Normal shutdown')
2024-12-05 13:52:49,254 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:52:49] "POST /api/sales/purchase HTTP/1.1" 200 -
2024-12-05 13:54:10,196 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:54:10] "GET /api/sales/history/jihad7 HTTP/1.1" 200 -
2024-12-05 13:55:53,113 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:55:53] "[35m[1mGET /api/sales/health HTTP/1.1[0m" 500 -
2024-12-05 13:57:07,398 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:57:07] "[35m[1mGET /api/sales/health HTTP/1.1[0m" 500 -
2024-12-05 13:57:23,497 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:57:23] "[35m[1mGET /api/sales/health HTTP/1.1[0m" 500 -
2024-12-05 13:58:49,534 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-05 13:58:49,534 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-05 13:59:01,965 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 13:59:01] "GET /api/sales/health HTTP/1.1" 200 -
2024-12-05 14:17:08,315 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-05 14:17:08,316 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-05 14:17:30,613 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:17:30] "[35m[1mPOST /api/inventory/ HTTP/1.1[0m" 201 -
2024-12-05 14:18:44,311 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:18:44] "[31m[1mGET /api/inventory/deduct HTTP/1.1[0m" 405 -
2024-12-05 14:19:05,117 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:19:05] "POST /api/inventory/deduct HTTP/1.1" 200 -
2024-12-05 14:20:13,927 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:20:13] "PUT /api/inventory/1 HTTP/1.1" 200 -
2024-12-05 14:21:11,646 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:21:11] "[35m[1mGET /api/inventory/health HTTP/1.1[0m" 500 -
2024-12-05 14:22:15,972 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:22:15] "[35m[1mGET /api/inventory/health HTTP/1.1[0m" 500 -
2024-12-05 14:22:30,531 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-05 14:22:30,532 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-05 14:22:33,116 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:22:33] "GET /api/inventory/health HTTP/1.1" 200 -
2024-12-05 14:26:02,388 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-05 14:26:02,388 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-05 14:30:01,800 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-05 14:30:01,800 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-05 14:30:05,161 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:30:05] "[31m[1mPOST /api/review/ HTTP/1.1[0m" 400 -
2024-12-05 14:37:37,495 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-05 14:37:37,496 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-05 14:37:40,374 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:37:40] "[35m[1mPOST /api/review/ HTTP/1.1[0m" 201 -
2024-12-05 14:39:34,258 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:39:34] "[31m[1mGET /api/review/1 HTTP/1.1[0m" 405 -
2024-12-05 14:39:46,233 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:39:46] "[31m[1mGET /api/review/1 HTTP/1.1[0m" 405 -
2024-12-05 14:40:03,129 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:40:03] "[31m[1mGET /api/review/1 HTTP/1.1[0m" 405 -
2024-12-05 14:40:12,722 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:40:12] "PUT /api/review/1 HTTP/1.1" 200 -
2024-12-05 14:40:45,173 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:40:45] "[35m[1mDELETE /api/review/1 HTTP/1.1[0m" 204 -
2024-12-05 14:41:06,453 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:41:06] "[35m[1mPOST /api/review/ HTTP/1.1[0m" 201 -
2024-12-05 14:41:21,541 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:41:21] "[35m[1mPOST /api/review/ HTTP/1.1[0m" 201 -
2024-12-05 14:41:48,121 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:41:48] "[35m[1mPOST /api/review/ HTTP/1.1[0m" 201 -
2024-12-05 14:42:06,723 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:42:06] "[35m[1mDELETE /api/review/2 HTTP/1.1[0m" 204 -
2024-12-05 14:42:35,391 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:42:35] "[35m[1mDELETE /api/review/1 HTTP/1.1[0m" 204 -
2024-12-05 14:42:48,648 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:42:48] "[33mDELETE /api/review/1 HTTP/1.1[0m" 404 -
2024-12-05 14:42:54,037 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:42:54] "[33mDELETE /api/review/2 HTTP/1.1[0m" 404 -
2024-12-05 14:43:06,368 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:43:06] "[31m[1mDELETE /api/review/product/1 HTTP/1.1[0m" 405 -
2024-12-05 14:43:15,476 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:43:15] "[31m[1mDELETE /api/review/product/1 HTTP/1.1[0m" 405 -
2024-12-05 14:44:27,804 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:44:27] "GET /api/review/product/1 HTTP/1.1" 200 -
2024-12-05 14:44:33,725 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:44:33] "GET /api/review/product/2 HTTP/1.1" 200 -
2024-12-05 14:44:41,942 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:44:41] "[35m[1mPOST /api/review/ HTTP/1.1[0m" 201 -
2024-12-05 14:44:56,031 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:44:56] "[35m[1mPOST /api/review/ HTTP/1.1[0m" 201 -
2024-12-05 14:45:07,107 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:45:07] "[35m[1mPOST /api/review/ HTTP/1.1[0m" 201 -
2024-12-05 14:45:13,178 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:45:13] "GET /api/review/product/1 HTTP/1.1" 200 -
2024-12-05 14:45:34,871 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:45:34] "[35m[1mPOST /api/review/ HTTP/1.1[0m" 201 -
2024-12-05 14:45:39,153 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:45:39] "GET /api/review/product/1 HTTP/1.1" 200 -
2024-12-05 14:46:23,030 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:46:23] "GET /api/review/user/2 HTTP/1.1" 200 -
2024-12-05 14:47:10,600 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:47:10] "[35m[1mGET /api/review/health HTTP/1.1[0m" 500 -
2024-12-05 14:48:09,365 - werkzeug - INFO - [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2024-12-05 14:48:09,366 - werkzeug - INFO - [33mPress CTRL+C to quit[0m
2024-12-05 14:48:11,750 - werkzeug - INFO - 127.0.0.1 - - [05/Dec/2024 14:48:11] "GET /api/review/health HTTP/1.1" 200 -
